### PR TITLE
wgengine: build for freebsd

### PR DESCRIPTION
**Pull request by wardn**
Signed-off-by: wardn <wardn@users.noreply.github.com>

### CODE REVIEW

File: wgengine/router_freebsd.go [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+	return &r
+}
+
+func (r *freebsdRouter) Up() error {
+	return nil
+}
+
+func (r *freebsdRouter) SetRoutes(rs RouteSettings) error {
+	if SetRoutesFunc != nil {
+		return SetRoutesFunc(rs)
```
Comment (danderson) [inline wgengine/router_freebsd.go]: Don't use SetRoutesFunc. It's used only by the mac/ios builds, as a way to bridge Go and Swift together. `router_freebsd.go` should implement Up() and SetRoutes() in terms of FreeBSD APIs or commands (e.g. `ifconfig`, `route`, or equivalent kernel APIs, whatever's idiomatic for the platform).

Comment (danderson) [reply wgengine/router_freebsd.go]: (to be clear, this is a bug on our end - SetRoutesFunc exists in the cross-platform code as a hack that enables us to build bits of the mac/ios code from Linux, to reduce the amount of breakage - but it's not clearly documented that this is what's happening)

**Comment by wardn** [comment]
The implementation for Up() and SetRoutes() is similar to #36, so I'll hold off on this request until that pull lands and verify that they're compatible.

**Comment by danderson** [comment]
Landed! It'd be cool if all the *bsd end up with compatible code, but don't be afraid to copy/paste for now. Once the per-platform support is fully mature we can start refactoring common stuff, but for right now "anything" is better than "perfect" :)

**Pull request merged**